### ISSUE TITLE
Observation bodies are full of backticks, so writing one through an unquoted heredoc executes them — and the file is written, and it is wrong

### ISSUE THREAD

**Issue by juniordurant** [rebelytics/one-skill-to-rule-them-all#149]
**Target:** `SKILL.md` — "How to Log" / "File format";
`references/observation-log.md` — the helper snippets.
**Verified at:** upstream `main` (`7518a85`), byte-identical to v3.2.0.
**Adjacent, not duplicate:** #117 argues log mutations should go through the
editing tool rather than `sed`/`perl`, for the same underlying reason (the
shell rewrites what you typed and exits 0). That issue is about *mutating an
existing* file with a corrupting command. This is about *creating* one, where
the shell is frequently the only available path — a `set -C` noclobber create
has no editing-tool equivalent — and where the corruption is not a stray byte
but silent command execution. #65 and #67 (post-write verification) are the
guard that would catch it; neither covers content of this kind.

## What happened

The documented flow is a noclobber create followed by writing the file. The
natural shell form for a multi-line Markdown body is a heredoc, and the
natural heredoc is unquoted:

```bash
cat >> "$f" <<OBS
**Issue:** The `find -exec` form spawns one process per file …
**Improvement:** use `$(…)` instead of backticks …
OBS
```

Every observation body this skill asks for is a technical body: it names
commands, fields and flags, and it does that in backticks and `$(…)` — the
skill's own template is written that way. In an unquoted heredoc, the shell
substitutes all of it. `` `find -exec …` `` runs. `$(…)` runs. `$f` expands.
The resulting file is created successfully, is valid Markdown, and says
something the author never wrote — usually with the command's *output*, or an
empty string, where the explanation used to be. Exit code 0.

It is durable, too. The frontmatter still parses, so the scan reports the file
as healthy; the damage is in the body, which nothing reads until a review
opens it weeks later.

## Why the current text did not prevent it

The skill has strong rules about *what* the file must contain — mandatory
frontmatter, `status: open`, a non-empty `siblings_checked:` — and about
*when* to write it. It says nothing about the mechanism of the write, so the
mechanism is improvised each time, and the improvisation that reads most
naturally is the unsafe one. `references/observation-log.md`'s helper snippets
are the obvious place for the safe form and currently stop short of the body.

## Suggested fix

Two sentences and a snippet in `references/observation-log.md`, with a pointer
from "How to Log":

> **Write the frontmatter with `printf`, the body with a QUOTED heredoc.**
> An observation body is technical prose full of backticks and `$(…)`, and an
> unquoted heredoc executes every one of them, writes the file, and exits 0.
> The delimiter must be quoted — `<<'OBS'` — which disables all substitution;
> that also means the body cannot carry variables, so the id and any dates go
> in the `printf` above it.
>
> ```bash
> printf -- '---\nid: %s\ntitle: "%s"\nstatus: open\n…\n---\n\n' "$next_id" "$title" > "$f"
> cat >> "$f" <<'OBS'
> **Issue:** … `find -exec` … $(date) … all written literally
> OBS
> ```
>
> Then re-read the file in the same turn: a body that came out shorter than
> you wrote it, or that contains command output, is this failure. Checking
> later means checking never.

The same rule covers `echo -e` and any `"$(cat <<…)"` variant. If #117's
"never write these files from the shell" is adopted, this becomes the
exception clause for the create path rather than a separate rule — the two
should probably land together.

---

Related (checked before filing, not duplicates): #117, #65, #67, #74.

---

Reported from a production install of v3.2.0 (20 project copies, a 200-entry
log). Prepared with Claude Code (Claude Opus 5); every command quoted above was
executed verbatim from a clean `sh` before posting, and a human reviewed and
approved this report.

**Comment by waylooking-hub** [comment]
A second, independent occurrence, and two refinements to the suggested fix that this case needed.

An observation file was written through an unquoted heredoc because a computed value had to appear in it. A command in backticks inside the body did not parse: the shell printed a syntax error, substituted nothing, and `cat` still exited 0. The next command in the `&&` chain ran, the file was created, a pre-commit hook passed and the commit went through. The damage was a gap in one sentence, found later by searching the file for the sentence it should have contained.

**1. "Re-read the file" misses this shape.** The suggested check looks for a body "shorter than you wrote it, or that contains command output". A failed or silent substitution produces neither in any visible way: the body loses a few words from one line and gains nothing, which a reader skims past. A mechanical form catches it: after writing, search the file with `grep -F` for a verbatim fragment that contains a backtick span, since that is exactly the text a substitution removes.

~~~bash
grep -F -q 'the loop form `for f in` splits' "$f" || echo "BODY DAMAGED: $f"
~~~

**2. A value that has to appear inside the body.** "The body cannot carry variables, so the id and any dates go in the `printf`" covers the header. When the body itself needs a computed value (the new file's own id in a cross-reference, a sibling id derived in the same command), write a placeholder inside the quoted heredoc and replace it in a second step that fails unless the placeholder occurs exactly once. An exact-match replacement in the editing tool does this, which also fits the direction of #117.

Checked from a clean `bash --noprofile --norc`: with an unquoted delimiter, a body line containing `` `for f in` `` came out as "the loop form  splits on spaces", `cat` returned 0, and the grep above reported the damage; a line containing `` `cd /` `` lost the span with no message at all, and the grep reported that too; with `<<'OBS'` the grep found the fragment.

Verified at `main` (`7518a85`): neither SKILL.md "How to Log" nor `references/observation-log.md` says anything about quoting a heredoc.

Prepared with Claude Code (Claude Opus 5) from observations logged while using this skill; filed on the account owner's instruction.

**Comment by rebelytics** [comment]
Thank you. In the queue and will make it into one of the next weekly releases.
